### PR TITLE
Created project layout and csv, conf readers

### PR DESCRIPTION
**Pull request by romitkarmakar**

### CODE REVIEW

File: include/CSVManager.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include<vector>
+#include<string>
+
+class CSVManager {
```
Comment (xcaptain) [inline include/CSVManager.h]: casbin doesn't read csv file directly, it uses an adapter to read policy, csv file is a kind of adapter.

File: include/CSVManager.h [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    std::vector<std::vector<std::string>> data;
+public:
+    std::vector<std::string> readLine(std::string);
+    void readFile(std::string);
+    void writeFile(std::string);
```
Comment (xcaptain) [inline include/CSVManager.h]: I think the `writeFile` api here has no use because we usually don't edit a csv file.

File: include/ConfReader.cc [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        return rld;
+    return "";
+}
+
+void ConfReader::readFile(std::string fileName)
+{
+    std::ifstream file(fileName, std::ios::out);
+    if (!file.is_open())
+    {
+        std::cerr << "Error: Unable to open CSV file " << fileName << " for reading!" << std::endl;
```
Comment (xcaptain) [inline include/ConfReader.cc]: conf file is actually an `ini` file, not a csv file. The purpose of a conf reader is to construct a model

File: include/ExpressionParser.cc [hunk 4]
```diff
@@ -0,0 +1,64 @@
+#include <vector>
+#include <string>
+#include <stack>
+#include <iostream>
+#include "utils.h"
+
+class ExpressionParser
```
Comment (xcaptain) [inline include/ExpressionParser.cc]: What's the purpose of this class?

File: makefile [hunk 5]
```diff
@@ -0,0 +1,4 @@
+CC = g++
+CFLAGS = -std=c++11
```
Comment (xcaptain) [inline makefile]: How about choosing a cross-platform build tool like cmake so we can build in windows.

**Comment by romitkarmakar** [comment]
@xcaptain thank you for your support, I will refactor the whole code according to your suggestions and code of conduct.

File: include/ExpressionParser.cc (hunk 4 above)
Comment (romitkarmakar) [reply include/ExpressionParser.cc]: This class is to parse the matcher expression passed in the model conf file

**Comment by romitkarmakar** [comment]
Added CMake and Visual Studio Support.

**Comment by hsluoyz** [comment]
See: https://github.com/casbin/casbin-cpp/issues/7

**Comment by romitkarmakar** [comment]
@hsluoyz  I will add support for makefile and Visual Studio project files support.

**Comment by hsluoyz** [comment]
I think contributors should add code based on my latest commits. Like start from: https://github.com/casbin/casbin-cpp/blob/master/casbin/enforcer.cpp

**Pull request closed without merging**